compiler/lexemes.py

```diff
@@ -21,18 +22,21 @@ def parse_for_variable_hierarchies(data, keyword):
         return data
 
     if keyword in data:
-        data = data[keyword]
+        if data_type is CommentedMap:
+            data = data[keyword]
+        else:
+            data = data[data.index(keyword)]
```

Comment:
Great thought on adding the data_type checking! 👍 
At present, however, the idea is that the __from__ keyword can be used as the only child of a CommentedMap object. The syntax looks as follows:

```
var2: &var2 value_2
var1: &var1
__from__: *var2
```
So we can ignore the case where ``` data = data[data.index(keyword)]``` would be required and let it throw an exception for such usage instead ;) 

Let's stick with just:
```
if keyword in data:
        data = data[keyword]
```
for now ;)

Reply:
Hi I have been thinking, in the test  data file I have used the below list structure. 
`nodes:
      - node: lw-site-droplet-0
        container_count: 1
      - node: ec2-18-184-37-92.eu-central-1
        container_count: 2`

But in order to use the DRY mechanism, we need to create a different variable name? If so we have to create a equal amount of field in the schema file instead of the following:

This can check every item in the list without typing repetitive component out.
`nodes: list(include('component'))
---
component:
   node: str()
   container_count: int()`